### PR TITLE
Added Rados missing polarion tests from test suites

### PR DESCRIPTION
**Pull request by pdhiran**
CEPH-83573526 - On PG merge, PGs get stuck in premerge state for extended periods of time

All test runs are on RHCS 7.1 latest.

CEPH-83573526 - On PG merge, PGs get stuck in premerge state for extended periods of time
Pass log: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-E4B45Y 

CEPH-83571453 - [RADOS]: corrupt snap info on primary osd and list-inconsistent-snapset in ec pool 
Bug : https://bugzilla.redhat.com/show_bug.cgi?id=2263023 

Fail log :  http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-KQYI62/Inconsistent_object_pg_check_for_EC_pools_0.log 

CEPH-83573413 - With pg auto balancing in warn mode, check whether it alerts for PG autoscaling
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1YH7G5

CEPH-83571452 - [RADOS]: corrupt snap info on NON-PRIMARY osd and list-inconsistent-snapset 
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-G7BPRE/

### CODE REVIEW

**Comment by neha-gangadhar** [comment]
Looks good. Test "CEPH-83571452" is not added to the suite. Can you please include it? The support is added but Polarion ID is missing.

**Comment by pdhiran** [comment]
> Looks good. Test "CEPH-83571452" is not added to the suite. Can you please include it? The support is added but Polarion ID is missing.

Added the missing ID from test suite.

File: ceph/rados/core_workflows.py [hunk 1]
```diff
@@ -419,7 +419,10 @@ def bench_write(self, pool_name: str, **kwargs) -> bool:
         max_objs = kwargs.get("max_objs")
         verify_stats = kwargs.get("verify_stats", True)
         check_ec = kwargs.get("check_ec", True)
-        cmd = f"sudo rados --no-log-to-stderr -b {byte_size} -p {pool_name} bench {duration} write --no-cleanup"
+        cleanup = kwargs.get("cleanup", False)
+        cmd = f"sudo rados --no-log-to-stderr -b {byte_size} -p {pool_name} bench {duration} write"
+        if cleanup:
```
Comment (harshkumarRH) [inline ceph/rados/core_workflows.py:424]: Please change the variable name to `nocleanup` or change the comparison to `if not cleanup`

File: conf/reef/rados/test-confs/pool-configurations.yaml [hunk 2]
```diff
@@ -14,7 +11,6 @@ replicated:
     pool_type: replicated
     pool_name: re_pool_30
     pg_num: 128
-    size: 3
```
Comment (harshkumarRH) [inline conf/reef/rados/test-confs/pool-configurations.yaml:17]: Please make changes to other versions of this file as well

File: suites/quincy/rados/tier-2_rados_test-pool-functionalities.yaml [hunk 3]
```diff
@@ -351,14 +351,17 @@ tests:
...
+  - test:
+      name: Verify cluster behaviour during PG autoscaler warn
+      module: pool_tests.py
+      polarion-id:  CEPH-83573413
+      config:
+        verify_autoscaler_warn:
+          pool_configs:
+            - type: replicated
+              conf: sample-pool-1
+          pool_configs_path: "conf/reef/rados/test-confs/pool-configurations.yaml"
```
Comment (harshkumarRH) [inline suites/quincy/rados/tier-2_rados_test-pool-functionalities.yaml:363]: Please change the pool_config path to quincy

File: suites/reef/rados/tier-2_rados_test-pg-split-merge.yaml [hunk 4]
```diff
@@ -281,3 +281,21 @@ tests:
...
+          - create_pool:
+              check_premerge_pgs: true
+              pool_name: pool3
+              pg_num: 32
+              rados_put: true
+              num_objs: 200
+              byte_size: 1024
+              pool_type: replicated
+        delete_pools:
+          - pool13
```
Comment (harshkumarRH) [inline suites/reef/rados/tier-2_rados_test-pg-split-merge.yaml:301]: Please change pool name to `pool3`

File: suites/pacific/rados/tier-2_rados_test-pool-functionalities.yaml [hunk 5]
```diff
@@ -351,14 +351,17 @@ tests:
...
+  - test:
+      name: Verify cluster behaviour during PG autoscaler warn
+      module: pool_tests.py
+      polarion-id:  CEPH-83573413
+      config:
+        verify_autoscaler_warn:
+          pool_configs:
+            - type: replicated
+              conf: sample-pool-1
+          pool_configs_path: "conf/reef/rados/test-confs/pool-configurations.yaml"
```
Comment (harshkumarRH) [inline suites/pacific/rados/tier-2_rados_test-pool-functionalities.yaml:363]: Please change pool config path to pacific

File: tests/rados/pool_tests.py [hunk 6]
```diff
@@ -511,26 +511,36 @@ def run(ceph_cluster, **kw):
         # Now that the autoscaler has been turned on, the warnings should go and PGs should be scaled accordingly
         time.sleep(300)
 
-        flag = True
-        end_time = datetime.datetime.now() + datetime.timedelta(seconds=2000)
+        alert_flag = False
+        end_time = datetime.datetime.now() + datetime.timedelta(seconds=4000)
```
Comment (harshkumarRH) [inline tests/rados/pool_tests.py:515]: Maybe we should reduce the wait time to 15 mins and monitor few runs, we can increase the wait time later on if observe false failures.

File: tests/rados/test_osd_snap_inconsistency_pg.py
Comment (harshkumarRH) [inline tests/rados/test_osd_snap_inconsistency_pg.py:1]: Please add a if-else block to create erasure coded pools and as OMAP entries are not possible for EC pools, maybe we can try writing normal objects using `rados bench`

File: ceph/rados/core_workflows.py (hunk 1 above)
Comment (pdhiran) [reply ceph/rados/core_workflows.py:424]: Done. Updated the key to match the param name.

File: conf/reef/rados/test-confs/pool-configurations.yaml (hunk 2 above)
Comment (pdhiran) [reply conf/reef/rados/test-confs/pool-configurations.yaml:17]: Updated the files

File: suites/quincy/rados/tier-2_rados_test-pool-functionalities.yaml (hunk 3 above)
Comment (pdhiran) [reply suites/quincy/rados/tier-2_rados_test-pool-functionalities.yaml:363]: Updated the suite files.

File: suites/reef/rados/tier-2_rados_test-pg-split-merge.yaml (hunk 4 above)
Comment (pdhiran) [reply suites/reef/rados/tier-2_rados_test-pg-split-merge.yaml:301]: updated the pool names.

File: suites/pacific/rados/tier-2_rados_test-pool-functionalities.yaml (hunk 5 above)
Comment (pdhiran) [reply suites/pacific/rados/tier-2_rados_test-pool-functionalities.yaml:363]: updated

File: tests/rados/pool_tests.py (hunk 6 above)
Comment (pdhiran) [reply tests/rados/pool_tests.py:515]: Updated.

File: tests/rados/test_osd_snap_inconsistency_pg.py
Comment (pdhiran) [reply tests/rados/test_osd_snap_inconsistency_pg.py:1]: Updated.

File: tests/rados/test_osd_snap_inconsistency_pg.py [hunk 7]
```diff
@@ -32,33 +32,67 @@ def run(ceph_cluster, **kw):
...
+                    return 1
+            else:
+                # Creating rados bench objects
+                if not rados_obj.bench_write(
+                    pool_name=pool_name,
+                    byte_size="1000KB",
+                    max_objs=50,
+                    rados_write_duration=30,
+                ):
+                    log.error(f"Omap entries not generated on pool {pool_name}")
```
Comment (pdhiran) [inline tests/rados/test_osd_snap_inconsistency_pg.py:79]: update error message to specify rados bench and no omap objects

**Review by harshkumarRH — DISMISSED** [review]
Looks good with all the changes.

**Review by SrinivasaBharath — APPROVED** [review]
LGTM

**Review by neha-gangadhar — APPROVED** [review]
lgtm

**Pull request merged**